tests/test_checker.py

```diff
@@ -69,7 +69,9 @@ def test_check_filtered(self):
 
         ext_data = dummy_checker.check()
         ext_data_from_getter = dummy_checker.get_external_data()
-        self.assertEqual(ext_data_from_getter, ext_data)
+        self.assertEqual(len(ext_data), len(ext_data_from_getter))
+        for data in ext_data:
+            self.assertIn(data, ext_data_from_getter)
```

Comment:
Why is this change needed – is the order no longer preserved?

Reply:
Yes, the order is not preserved on the return value of `ManifestChecker.check()` anymore. That is because we run the modules check first and add all modules external data to the check list and then we check individual sources that were not processed yet and add them as they get processed (opposed to only checking sources in order as it was before).

As far as I could see nothing (apart from this test) was using the return value of `ManifestChecker.check()`, so I think it should be safe but please let me know.

Reply:
It's probably fine but perhaps you could express this as:

```
         self.assertEqual(len(ext_data), len(ext_data_from_getter))
         self.assertEqual(set(ext_data), set(ext_data_from_getter))
```

?

Reply:
Done.